### DISCUSSION TITLE
Condition block fails, symbol primitive vs. complex

### DISCUSSION THREAD

**Discussion by littlefixit** [postmanlabs/postman-flows#268] [Q&A]
Attempting to perform a simple String comparison:
```
/data/body/value/0/CardType == "cCustomer"
```

Encountering the following error:

Value "/body/value/0" is a "symbol" primitive. Must be "complex" to have field "CardType"

Cannot locate documentation on this functionality--please advise. Thanks!

**Comment by saswatds** [comment]
Hey @littlefixit, this is bug in our implementation. This also related to the same problem as #255

**Reply by saswatds** [reply]
Hey @littlefixit, I have fixed the issue and will release it soon but before that I just want to add a test to ensure this does not happen in future. Would you be willing to share your JSON object or the schema of the object (Sensitive information replaced with fake data)?

**Comment by littlefixit** [comment]
For anyone else struggling with this:

the solution is to introduce an iterator between the output of the request and piping the response data through an Iterator. The output of the iterator can then be fed into the conditional check, and everything will work as expected.

Hopefully, the code can be fixed to eliminate the need for this eventually, but in the meanwhile this is a passable workaround.
